functests/utils/cluster/cluster.go

```diff
@@ -0,0 +1,38 @@
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// EnforceRequirements indicates if tests with specific cluster preconditions should fail rather than skip
+// if the cluster we are running against doesn't meet the requirements of the tests
+func EnforceRequirements() bool {
+	enforceReqs, _ := strconv.ParseBool(os.Getenv("ENFORCE_REQUIREMENTS"))
+	return enforceReqs
+}
+
+func HaveEnoughCores(nodes []corev1.Node, cpus int) (bool, error) {
+	enoughCores := false
```

Comment:
the intent of the code is to report no valid nodes unless proven otherwise (at least 1 suitable node found), if this is not clear enough I need to improve it, please feel free to suggest what confused you.

Reply:
mostly I meant to remove the variable and have something like 
```
func HaveEnoughCores(nodes []corev1.Node, cpus int) (bool, error) {
	requestCpu := resource.MustParse(fmt.Sprintf("%dm", cpus*1000))
	for _, node := range nodes {
		availCpu, ok := node.Status.Allocatable[corev1.ResourceCPU]
		if !ok {
			return false, fmt.Errorf("cpu resource not in allocatable on node %q", node.Name)
		}
		if availCpu.IsZero() {
			return false, fmt.Errorf("zero available cpu on node %q", node.Name)
		}

		if availCpu.Cmp(requestCpu) >= 1 {
			return true, nil
		}
	}

	return false, nil
}
```

Reply:
done